packages/core/contracts-ovm/insured-bridge/implementation/BridgePool.sol

```diff
@@ -159,36 +163,38 @@ contract BridgePool is Testable, BridgePoolInterface, ExpandedERC20 {
      * @param depositId Unique ID corresponding to deposit order that caller wants to relay.
      * @param depositTimestamp Timestamp of Deposit emitted by L2 contract when order was initiated.
      * @param recipient Address on this network who should receive the relayed deposit.
-     * @param amount Deposited amount.
-     * @param realizedFeePct Computed offchain by caller, considering the amount of available liquidity for the token
-     * currency needed to pay the recipient and the count of pending withdrawals at the `depositTimestamp`. This fee
-     * will be subtracted from the `amount`. If this value is computed incorrectly, then the relay can be disputed.
-     * @param maxFeePct Maximum fee that L2 Depositor can pay. `realizedFee` <= `maxFee`.
-     * @param proposerRewardPct Reward % of deposit amount to pay relayers.
+     * @param amount Amount deposited on L2 to be brought over to L1.
+     * @param slowRelayFeePct Max fraction of `amount` that the depositor is willing to pay as a slow relay fee.
+     * @param instantRelayFeePct Fraction of `amount` that the depositor is willing to pay as a instant relay fee.
+     * @param quoteDeadline Timestamp up until the depositor is willing to accept an LP quotation for.
+     * @param realizedLpFeePct LP fee calculated off-chain considering the L1 pool liquidity at deposit time, before
+     *      quoteDeadline. The OO acts to verify the correctness of this realized fee. Can not exceed 50%.
      */
     function relayDeposit(
         uint64 depositId,
         uint64 depositTimestamp,
         address recipient,
         address l2Sender,
         uint256 amount,
-        // TODO: Allow caller to distinguish between slow and fast fees/rewards.
-        uint64 realizedFeePct,
-        uint64 maxFeePct,
-        uint64 proposerRewardPct
+        uint64 slowRelayFeePct,
+        uint64 instantRelayFeePct,
+        uint64 quoteDeadline,
+        uint64 realizedLpFeePct
     ) public {
-        require(realizedFeePct <= maxFeePct, "Invalid realized fee");
+        require(realizedLpFeePct <= 0.5e18, "Invalid realized LP fee");
```

Comment:
Should we be doing max(fee, 0.5) instead of require? Also, should we be validating that the other fees are <= 0.25 here too to ensure this method always works?

Reply:
yup. capping the fee at 0.5 to prevent reverts sounds like a good idea. Validating the other fees here is also possible. if this was not done here anything that exceeds this should be disputed but I agree it's good idea to cap it to prevent the dispute state.

Reply:
Other argument could be that under no saituations should this value ever be greater than 0.5e18 and so preventing revert might be better than a ceil.

Reply:
After syncing on this we agreed that a `require` is sufficient.